clang/test/Driver/sycl-oneapi-gpu-intelgpu.cpp

```diff
@@ -287,13 +286,10 @@
 // RUN:   -fno-sycl-device-lib=all -fno-sycl-instrument-device-code \
 // RUN:   -target x86_64-unknown-linux-gnu -### %s 2>&1 | \
 // RUN:   FileCheck %s --check-prefix=CHECK_TOOLS_BEOPTS_MIX
-// CHECK_TOOLS_BEOPTS_MIX: ocloc{{.*}} "-device" "dg1"{{.*}}"-DDG1"
 // CHECK_TOOLS_BEOPTS_MIX: opencl-aot{{.*}} "-DCPU"
-// CHECK_TOOLS_BEOPTS_MIX-NOT: "-DDG1"
-// CHECK_TOOLS_BEOPTS_MIX: ocloc{{.*}} "-device" "skl"{{.*}}"-DSKL2"
+// CHECK_TOOLS_BEOPTS_MIX: ocloc{{.*}} "-device" "dg1,skl"{{.*}}"-DDG1" "-DSKL2"
 
 /// Check that target is passed to sycl-post-link for filtering
 // RUN: %clangxx -fsycl -fsycl-targets=intel_gpu_pvc,intel_gpu_dg1 \
 // RUN:   -### %s 2>&1 | FileCheck %s --check-prefix=CHECK_TOOLS_FILTER
-// CHECK_TOOLS_FILTER: sycl-post-link{{.*}} "-o" "intel_gpu_pvc,{{.*}}"
-// CHECK_TOOLS_FILTER: sycl-post-link{{.*}} "-o" "intel_gpu_dg1,{{.*}}"
+// CHECK_TOOLS_FILTER: sycl-post-link{{.*}} "-o" "intel_gpu_pvc,dg1{{.*}}"
```

Comment:
The [design](https://github.com/intel/llvm/pull/12252/files#diff-a6ead7573259631ce08e0d9f54439282456c4a66431e8a39f9a4177fc81a3e77R992-R995) for optional kernel features AOT had the driver invoking `sycl-post-link` with `-o <device>,<output.table>` for each target. This looks like it's invoked with one `-o` and with all the targets (plus the output file) listed in one argument, while `sycl-post-link` expects the value of `-o` to be a `<device>,<output>` pair. Is it possible/are you planning to update the driver to invoke `sycl-post-link` with the multiple `-o` format?

Reply:
@jzc, thanks for pointing that out.  This would indicate that this current solution is not complete and only provides the step to have a single sycl-post-link call.  Additional work is needed to call `sycl-post-link` with multiple outputs, one for each target specified by the user, which subsequently would perform multiple `ocloc` calls.

@asudarsa, FYI, for work needed for the new offloading model.

Reply:
I don't see a strong technical reason for calling `sycl-post-link` multiple times, except that it will require `sycl-post-link` revamp to handle multiple targets in one invocation.

Considering that `sycl-post-link` is going to be consumed by `clang-linker-wrapper` in the new offloading model, I suggest that we follow the currently proposed design for `sycl-post-link`, but make sure that `clang-linker-wrapper` is only called just once regardless of how many targets we specify.